notebooks/rocoto.ipynb

```diff
@@ -0,0 +1,1013 @@
+    "    output_file='tmp/simple-workflow.xml'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12b07a2c-7f9c-4480-b165-9e3b5a0ced6d",
+   "metadata": {},
+   "source": [
+    "The resulting Rocoto XML file is shown below. An XML header is automatically added without the need to explicitly define it in the UW YAML. Note how blocks from UW YAML language have been transformed into XML tags along with their attributes and values. For example, attributes defined by`attrs:` block in the UW YAML have become attributes within the `<workflow>` tag in the XML file. \n",
+    "\n",
+    "For more information on Rocoto workflows, including tags like the ones shown here and thier attributes, see the <a href=\"https://christopherwharrop.github.io/rocoto/\">Rocoto Documentation</a>.\n",
+    "<!--cell 8-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "d895e1cf-e8af-437b-9a38-2b03ec34f527",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<?xml version='1.0' encoding='utf-8'?>\n",
+      "<workflow realtime=\"False\" scheduler=\"slurm\">\n",
+      "  <cycledef>202410290000 202410300000 06:00:00</cycledef>\n",
+      "  <log>logs/test.log</log>\n",
+      "  <task name=\"greet\">\n",
+      "    <cores>1</cores>\n",
+      "    <walltime>00:00:10</walltime>\n",
+      "    <command>echo Hello, World!</command>\n",
+      "    <jobname>greet</jobname>\n",
+      "  </task>\n",
+      "</workflow>\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat tmp/simple-workflow.xml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "39f50dbd-e3f3-4e1e-a10b-f622f635b16b",
+   "metadata": {},
+   "source": [
+    "The following is a workflow that is missing required components. `workflow` doesn't contain a `realtime` attribute, a `log:` block isn't included, and `task_greet` doesn't include a `command`.\n",
+    "<!--cell 10-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "56d04e6b-5fa2-4c93-ac2f-c95fa23c888e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "workflow:\n",
+      "  attrs:\n",
+      "    scheduler: slurm\n",
+      "  cycledef:\n",
+      "    - spec: 202410290000 202410300000 06:00:00\n",
+      "  tasks:\n",
+      "    task_greet:\n",
+      "      cores: 1\n",
+      "      walltime: 00:00:10\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/rocoto/err-workflow.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f102c6ca-c8a5-4f9c-8e40-b5fa040bbeab",
+   "metadata": {},
+   "source": [
+    "When any type of validation error occurs, `realize()` will raise an exception indicating what type of error occurred. The number of validation errors present and their locations within the workflow structure are given.\n",
```

Comment:
```suggestion
    "When validation errors occur, `realize()` raises an exception indicating what type of error occurred. If logging is configured, the number of validation errors and their locations within the workflow structure are also shown.\n",
```

Reply:
I like most of the changes here, but when I test this the details on validation errors are shown even without logging configured.

Reply:
I think (as mentioned in another chain of comments somewhere here) we can probably just leave out the language about the logger, as it's not really pertinent. Good catch, too.